### DISCUSSION TITLE
클러스터 ClusterMarkerProp의 추가 지원

### DISCUSSION THREAD

**Discussion by jun-jaehyuk-lee** [mj-studio-library/react-native-naver-map#81] [Q&A]
좋은 라이브러리 사용 도입을 위해 react-native 0.74.3 최신버전의 fabric을 이용해 테스트 중에 있습니다.

현재 클러스터의 ClusterMarkerProp 의 width, height 가 클러스터의 마커 이미지의 사이즈뿐 아니라 클러스터의 사이즈도 같이 영향을 받고있습니다.

이에 클러스터의 style props와 클러스터 마커의 style props 를 따로 분리해 커스텀할 수 있는 기능이 필요합니다. (클러스터의 색상 및 클러스터의 숫자 text의 font 및 사이즈,컬러 수정이 된다면 더욱 좋겠습니다.)

좋은 라이브러리 사용 기여에 감사드립니다.

@mym0404

**Comment by mym0404** [comment]
혹시 어떻게 되는건지 사진첨부 가능할까요? Cluster marker에 대한 여러 프로퍼티 추가 계획은 있습니다.

--

직접 구동시켜서서 확인해보았습니다. 이렇게 cluster marker가 leaf marker의 크기에 영향을 받는다는 의미인가요? 
![image](https://github.com/mym0404/react-native-naver-map/assets/33388801/4149d1e2-7fae-44b2-b10d-ac6ef30bf429)

**Reply by jun-jaehyuk-lee** [reply]
![cluster](https://github.com/mym0404/react-native-naver-map/assets/100741856/7ed937f5-498a-422a-8126-dc0ef388aae9)

**Reply by jun-jaehyuk-lee** [reply]
클러스터의 원 숫자표기되는 부분의 스타일과 클러스터가 일정 줌레벨에 도달했을때 보이는 cluster marker의 스타일입니다.
현재 첨부드린 UI는 NaverMapMarkerOverlay의 마커는 사용하지 않은 상태입니다.

**Reply by jun-jaehyuk-lee** [reply]
cluster marker가 leaf marker의 크기에 영향을 받는다는 의미인가요?
cluster marker(원 및 원에 표기되는 숫자) leaf marker(클러스터가 특정 줌레벨에 도달했을때 onTapClusterLeaf를 사용하는 마커) 라면 말씀하신 부분이 맞습니다.

**Reply by mym0404** [reply]
확인했습니다. 네이버맵 내부적으로 클러스터의 크기를 지정해주지 않으면 리프의 크기를 따르게되는(그것고 어떤건 따르고 어떤건 따르지않은) 버그 비슷한게 있는것같은데, 클러스터에도 크기를 전달할수 있는 방향으로 기능 추가가 필요해보입니다.

**Reply by jun-jaehyuk-lee** [reply]
네 답변 감사드립니다.
바쁘실텐데 클러스터의 크기뿐아니라 backgroundColor, fontSize, color .... 기타 style props도 추가하는데 이슈가 없다면 도입이 되면 좋을거같습니다.🙏
좋은 라이브러리 사용 기여에 감사드립니다.

**Reply by mym0404** [reply]
네 확인했습니다. 후원 감사드립니다. 열심히 반영해보도록 하겠습니다. #89

**Reply by jun-jaehyuk-lee** [reply]
감사합니다. 이건 라이브러리 외적으로 개발중이던 앱을 fabric으로의 전환을 위해 테스트를 진행중인데 fastImage 부터 lottie 등 라이브러리 이슈들이 끊이질 않네요. 지금 도입하는게 맞는지가 의문이네요ㅜㅜ

**Comment by mym0404** [comment]
`2.1.1`에 cluster 에 `width`, `height` 및 이걸 지정하지 않아도 크기가 leaf marker의 크기를 따라가지 않는 기능을 추가하여 기능 업데이트 되었습니다. 나머지 옵션들은 다음 업데이트때 추가 예정입니다.

**Reply by jun-jaehyuk-lee** [reply]
업데이트 감사합니다.
약 일주일간 react-native fabric으로 여러가지 테스트를 해보았는데, 선택적 라이브러리가 아닌 필수적인 라이브러리들에서 조차도 이슈들이 너무 많다는 판단을 내렸습니다. 결국 구 아키텍쳐와 1.5.9 버전으로 다시 rollback을 했네요, 1.5.9버전에 onTapClusterLeaf 및 해당 기능들이 포함되기를 열망하는 1인입니다..
긴글 읽어주셔서 감사합니다.
